src/contexts/W3iContext/hooks/w3iProxyHooks.ts

```diff
@@ -28,8 +28,13 @@ export const useW3iProxy = () => {
   )
 
   useEffect(() => {
-    w3iProxy.init().then(() => setReady(true))
-  }, [w3iProxy, setReady])
+    if (w3iProxy.isInitializing) {
+      return
+    }
+    if (!w3iProxy.getInitComplete()) {
+      w3iProxy.init().then(() => setReady(true))
```

Comment:
While on the note of explicit error handling:

```suggestion
      w3iProxy.init().then(() => setReady(true)).catch(error => { console.error("w3iProxy failed to initialize: ", error) })
```